timelock-server/src/suiteTest/java/com/palantir/atlasdb/timelock/MultiNodePaxosTimeLockServerIntegrationTest.java

```diff
@@ -193,4 +197,29 @@ public void canCreateNewClientsDynamically() {
         }
     }
 
+    @Test
+    public void lockRequestCanBlockForTheFullTimeout() {
+        // Test proxy timeout is 12.5 seconds, so 15 seconds suffices.
+        LockToken token = client.lock(LockRequest.of(LOCKS, DEFAULT_LOCK_TIMEOUT_MS)).getToken();
+
+        try {
+            LockResponse response = client.lock(LockRequest.of(LOCKS, 15_000));
+            assertThat(response.wasSuccessful()).isFalse();
+        } finally {
+            client.unlock(token);
+        }
+    }
+
+    @Test
+    public void waitForLocksRequestCanBlockForTheFullTimeout() {
+        // Test proxy timeout is 12.5 seconds, so 15 seconds suffices.
```

Comment:
bit obscure, is there a way we can refer to `ClientOptions.NON_BLOCKING_READ_TIMEOUT.plus(Duration.ofSeconds(1))` as the lock timeout here?

Reply:
as an aside, whilst trying to find that option, it's unclear to me why we have `clientQoS` enabled as a default, why is that?

Reply:
Yep, I'm OK with making that public.

ClientQoS: I don't know as well, other than that the normal default for that in most uses outside of Atlas should be enabled.